internal/pkg/api/error.go

```diff
@@ -518,13 +520,26 @@ func (er HTTPErrResp) Write(w http.ResponseWriter) error {
 func ErrorResp(w http.ResponseWriter, r *http.Request, err error) {
 	zlog := hlog.FromRequest(r)
 	resp := NewHTTPErrResp(err)
-	e := zlog.WithLevel(resp.Level).Err(err).Int(ECSHTTPResponseCode, resp.StatusCode)
+	e := zlog.WithLevel(resp.Level).Err(err).Int(ECSHTTPResponseCode, resp.StatusCode).Str("error.type", fmt.Sprintf("%T", err))
 	if ts, ok := logger.CtxStartTime(r.Context()); ok {
 		e = e.Int64(ECSEventDuration, time.Since(ts).Nanoseconds())
 	}
 	e.Msg("HTTP request error")
 
 	if resp.StatusCode >= 500 {
+		if trans := apm.TransactionFromContext(r.Context()); trans != nil {
+			esErr := &es.ErrElastic{}
+			if errors.As(err, &esErr) {
+				trans.Context.SetLabel("error.type", "ErrElastic")
+				trans.Context.SetLabel("error.details.status", esErr.Status)
+				trans.Context.SetLabel("error.details.type", esErr.Type)
+				trans.Context.SetLabel("error.details.reason", esErr.Reason)
+				trans.Context.SetLabel("error.details.cause.type", esErr.Cause.Type)
+				trans.Context.SetLabel("error.details.cause.reason", esErr.Cause.Reason)
```

Comment:
@joshdover is this closer to what you had in mind (in the issue description)?

Reply:
Yes this makes sense to me. One thing I can't tell from just reading this is if any of these fields contain anything instance-specific content that will create noise and make it harder to group related errors, such as specific disk usage stats, node names, etc. This would probably be tricky to normalize though at this point and I think we can do this at query time later if it's an issue.

Reply:
ok, we'll merge as is for now and we can revist/clean up the labels in the future